### DISCUSSION TITLE
How does routing work?

### DISCUSSION THREAD

**Discussion by han-so1omon** [ExistentialAudio/BlackHole#634] [Q&A] [answered]
Is it that each output channel goes directly into the corresponding input channel? Or is it something different? I can't find any information on this. It would be very helpful for this to be explained somewhere. Thanks!

Here is how I think of it currently, which would be a decent diagram if it is correct:

Output 1 -> Input 1
Output 2 -> Input 2
Output 3 -> Input 3
...

**Comment by orchetect** [comment, marked as answer]
BlackHole is a loopback driver. So yes, each output passes audio to its corresponding input. Just like IAC MIDI does for MIDI.

**Reply by orchetect** [reply]
The main README has been updated to describe BlackHole as a loopback driver which should clarify this.
